pkg/kinds/general.go

```diff
@@ -124,6 +134,56 @@ func (m *GrafanaResourceMetadata) SetSlug(v string) {
 	m.set(annoKeySlug, v)
 }
 
+func (m *GrafanaResourceMetadata) GetTitle() string {
+	return m.Annotations[annoKeyTitle]
+}
+
+func (m *GrafanaResourceMetadata) SetTitle(v string) {
+	m.set(annoKeyTitle, v)
+}
+
+func (m *GrafanaResourceMetadata) GetDescription() string {
+	return m.Annotations[annoKeyDescription]
+}
+
+func (m *GrafanaResourceMetadata) SetDescription(v string) {
+	m.set(annoKeyDescription, v)
+}
+
+func (m *GrafanaResourceMetadata) GetTags() []string {
+	var tags []string
+	v, ok := m.Annotations[annoKeyTags]
+	if ok {
+		err := json.Unmarshal([]byte(v), &tags)
+		if err != nil && len(v) > 0 {
+			return strings.Split(v, ",")
+		}
+	}
+	return tags
+}
+
+func (m *GrafanaResourceMetadata) SetTags(tags []string) {
+	str := ""
+	if len(tags) > 0 {
+		hasComma := false
+		for _, t := range tags {
+			if strings.Contains(t, ",") {
+				hasComma = true
+				break
+			}
+		}
+		if hasComma {
+			out, err := json.Marshal(tags)
+			if err == nil {
+				str = string(out)
+			}
+		} else {
+			str = strings.Join(tags, ",")
+		}
+	}
```

Comment:
this seems a bit weird, is there any reason to support commas in tags?

Reply:
because we do today :(

Reply:
but now is a good time to change that... since yes it is weird.  I updated this so we replace any comma with a "-" and then clients can just split